### PR TITLE
Adds Radio Jammers

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Adds radio jammers, a TG port: https://github.com/tgstation/tgstation/pull/24667

Radio jammers are a traitor device that screws up radio signals badly if you're in range of one; it costs 5 TC and has a range of 12; it won't stop your target from yelling or calling for help, but it'll make it difficult to pinpoint where they're calling for help at.

img

:cl: Fox McCloud
add: Adds in radio jammers to the traitor uplink
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -146,3 +146,22 @@ effective or pretty fucking useless.
...
+	icon_state = "jammer"
+	var/active = FALSE
+	var/range = 12
+
+/obj/item/device/jammer/Destroy()
+	active_jammers -= src
+	return ..()
+
+/obj/item/device/jammer/attack_self(mob/user)
+	to_chat(user,"<span class='notice'>You [active ? "deactivate" : "activate"] the [src]<span>")
```
Comment (SamHPurp) [inline code/game/objects/items/devices/traitordevices.dm]: your closing span doesn't close, its missing the `/`

also does it need the `the`?

Missing full stop/period.

Comment (Fox-McCloud) [inline code/game/objects/items/devices/traitordevices.dm]: That's an operator, not a question, so yes, yes it does.

Comment (SamHPurp) [inline code/game/objects/items/devices/traitordevices.dm]: Fair enough. Other two points remain however.

**Comment by scrubmcnoob** [comment]
😨 that call for help gave me legit PTSD 😭 

Like the idea though.

**Comment by Spartan6** [comment]
Sprite for radio jamming item?

**Comment by Vivalas** [comment]
image

**Comment by Anticept** [comment]
This would be the legit counter against PAI death alarms.

**Comment by Fox-McCloud** [comment]
>That said, will this jam death alarms?

It won't; I can make it so so though--our radio system isn't as unified as TG's is, so we have multiple radio procs.

**Comment by Fox-McCloud** [comment]
Should work with death alarms pet collars, brig doors, and beepsky now.

File: code/game/objects/items/weapons/implants/implant_death_alarm.dm [hunk 2]
```diff
@@ -35,7 +35,7 @@
 	var/mob/M = imp_in
 	var/area/t = get_area(M)
 
-	var/obj/item/device/radio/headset/a = new /obj/item/device/radio/headset(null)
+	var/obj/item/device/radio/headset/a = new /obj/item/device/radio/headset(src)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/implants/implant_death_alarm.dm]: had to be done; can't get a turf of something in null space.

**Comment by Fruerlund** [comment]
Hmm I like it a lot. Cool feature.

The only problem I have with is if you are using this while assaulting someone and your target attempts to use his radio. The "jammed" text will give it away that they are being jammed as a result of the format/text style etc. 

Are there other alternatives to use instead or a font that isn't such a give away?

**Comment by Fox-McCloud** [comment]
I think that's the whole point. It's not meant to be a catch-all counter where you can just use it and you suffer zero chances of retaliation.

It gives you that precious time to murder someone and them not to be to say "; help I'm in sci maint near the airlock; he's trying to kill me."

It's not meant to be "I use this on you and you have no chance to call out for help or get it, ever"

**Comment by tigercat2000** [comment]
We can always tweak these later after seeing how they do.

**Pull request merged**